### ISSUE TITLE
Items which lose durability through crafting

### ISSUE THREAD

**Issue by Zelly9191** [crazysnailboy/UncraftingTable#6]
When uncrafting things that reqire items that loose durability through crafting such as the chisel from tinkers construct or the saw from microblocks they give a chizel/saw back. This was not a problem before the empty map fix I have a hunch that changed this problem. It use to just say uncrafting was impossible.

**Comment by crazysnailboy** [comment]
Which version of the mod does this apply to, and how can i reproduce it?

I've tried uncrafting a wooden chisel from TC using 1.4.2 and it still tells me it's possible.

**Comment by Zelly9191** [comment]
Try uncrafting smeltery brick blocks (black seared brick) in 1.7.10 you should get a chisel back. If not i can test some more to figure out why it happened to me.

**Comment by Nevik9191** [comment]
1.7.4 my bad.

**Comment by Zelly9191** [comment]
Wow I can't believe I did not catch that enchanted book thing. I feel very dumb right now! That was the one thing I forgot to test.

**Comment by crazysnailboy** [comment]
Forgive me if I'm misunderstanding this because I'm not very familiar with TC.

To craft TC chiseled seared bricks, for example, you place a seared paver block and a chisel into the crafting table. This then gives you a chiseled brick as the crafting output, and reduces the durability on the chisel.

When uncrafting a chiseled seared brick, it returns the chiseled paver but no chisel. 

This seems correct to me? The chisel isn't part of the crafting recipe exactly, it's just something that TC requires to be present while the crafting happens. If a chisel was included in the uncrafting result, wouldn't this be a source of free chisels as you get more from the uncraft than you do from the craft?

**Comment by Zelly9191** [comment]
To craft seared brick blocks there are 2 methods one is to use bricks. The other is to chisel seared pavers. When you chisel a block (seared pavers) you put the block and a chisel in a crafting table. The output will remove the block from the crafting grid and place a newly chiseled block (seared bricks) in the crafting output. Meanwhile the chisel has been left alone in the crafting table aside from losing some durability when chiseling the input (seared pavers) with it.

What should happen is the uncrafting table should check if the uncrafting table should check if a stagnant item is required to make this object. (like getting buckets back but using the milk when making a cake. in this case the full bucket is a chisel with whatever durability it had before and the empty bucket is the end durability chisel.) And should require the item to be in the calculus slot. (to return milk or durability to when uncrafting). 

CURRENTLY what happens is you uncraft a cake (or seared brick block) you duplicate the buckets (or chisel (which comes back as an invisible object in your inventory)) you would now have the empty buckets and the duplicated full buckets again. (chisel you used and invisible chisel created). 

Sorry i'm doing my best to explain it. If you don't get it let me know and ill try to show you either with a video (very unlikely) or screen shots. I'm difficult to understand sometimes. So just let me know what you do and don't get.

**Comment by Zelly9191** [comment]
Any luck so far?

**Comment by crazysnailboy** [comment]
Not really. I've had a look at how Minecraft handles the cake recipe and it has this concept of container items which I can probably use to force the user to put empty buckets in the compute slot and consume them as it creates milk buckets. I don't think the TC chisel is handled in the same way though so it's probably going to be pretty difficult to come up with something which is suitably generic.

**Comment by Zelly9191** [comment]
So I did some thinking and I know what you mean. My first thought is to have the uncrafting table do a check for any item(s) that remain in the crafting grid, during the completed crafting of the item that is intended to be uncrafted. If that makes any sense.

If this is possible it will allow this to work for items like liquid containers and for any other items as well. ex: full batteries that are required for a recipe end up drained in the crafting grid after.

The reason the tinkers construct chisel recipes are different probably has to do with the fact that all tinkers construct chisels share the same id and metadata regardless of material. To get around this the uncrafting table should check if a chisel from tinkers construct is required for the recipe it has found and if so search for a different recipe. If none are found then uncrafting should be declared impossible for said item.

let me know what you think. I'm thinking of making an animation that can play when the uncraft button is moused over if you would be up for it as well.

**Comment by crazysnailboy** [comment]
I think I've made some progress with this - I can identify items which have container items like buckets in the crafting recipe, so I'm going to try and make it so that the user has to add the container items to do the uncraft - either in the book slot or in the uncrafting grid, not sure yet. I haven't specifically checked the TC chisel yet, but in the process of adding IC2 support I discovered that IC2 recipes that require tools treat the tools as container items, so there's a good chance TC will as well.

Check out the new screenshots I added to the forum thread if you've not seen them already, they give an idea of the direction I'm heading in. At the moment I don't have an uncrafting button as I want to make it so that taking items out of the grid does the uncrafting, but if I have to include a button to handle container items then I'll let you know.

I was wondering if you'd be up for making some new textures? I'll make a new ticket for that so that this thread stays on topic :)
